File: shopeasy/src/main/java/com/shopeasy/service/AddressService.java
package com.shopeasy.service;

import java.util.List;

import com.shopeasy.Dto.AddressDTO;

public interface AddressService {

    AddressDTO addAddress(String email, AddressDTO dto);

    List<AddressDTO> getAllAddresses(String email);

    AddressDTO getAddressById(String email, Long id);

    AddressDTO updateAddress(String email, Long id, AddressDTO dto);

    void deleteAddress(String email, Long id);

}

File: shopeasyfrontend/src/services/AddressService.js
import api from "./api";

export const getAllAddresses = async () => {

    const response = await api.get("/address");

    return response.data;

};

export const addAddress = async (address) => {

    const response = await api.post("/address", address);

    return response.data;

};

export const updateAddress = async (id, address) => {

    const response = await api.put(`/address/${id}`, address);

    return response.data;

};

export const deleteAddress = async (id) => {

    const response = await api.delete(`/address/${id}`);

    return response.data;

};